taichi/program/state_flow_graph.cpp

```diff
@@ -861,6 +874,7 @@ std::vector<TaskLaunchRecord> StateFlowGraph::extract_to_execute() {
     // The reason we do this is that, upon the next launch, we could insert
     // edges to these executed but retained nodes. To allow for insertion, we
     // have to "unsort" them here.
+    nodes_[i]->input_edges.unsort_edges();
```

Comment:
Why do we need to unsort input edges?

Reply:
`sort_node_edges()` will work on both input and output edges, so we had to unsort them both. (This wasn't a problem when using `rebuild_graph()`, because `rebuild_graph()` does a clear first..)